scipy/stats/_stats_py.py

```diff
@@ -1099,12 +1099,13 @@ def _moment(a, moment, axis, *, mean=None):
     if a.size == 0:
         return np.mean(a, axis=axis)
 
-    if moment == 0 or moment == 1:
+    dtype = a.dtype.type if a.dtype.kind in 'fc' else np.float64
+
+    if moment == 0 or (moment == 1 and mean is None):
```

Comment:
Please check me here.

Reply:
> Please check me here.

What do you mean? check that this makes sense?

Reply:
Yes

Reply:
It seems to me that in any case, `_moment` is never called with `mean=None`, because `moment` always precalculates the mean for `_moment`.

I merged the suggested changes here locally, without this change. We can discuss it later on https://github.com/scipy/scipy/pull/18152.